Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39706&attempt_id=5aca9824-b10e-4f3f-b736-21251198c009)

Attempted: ocaml

<details><summary>Partial log (click to expand)</summary><p>

```
Error: Error during linking
make[3]: *** [../Makefile:74: unix.cmxs] Error 2
make[3]: Leaving directory '/build/ocaml-4.04.2/otherlibs/unix'
make[2]: *** [Makefile:680: otherlibrariesopt] Error 2
make[2]: Leaving directory '/build/ocaml-4.04.2'
make[1]: *** [Makefile:187: opt.opt] Error 2
make[1]: Leaving directory '/build/ocaml-4.04.2'
make: *** [Makefile:49: world.opt] Error 2
builder for '/nix/store/ink9f6z9z231c4iwcgkbxnfpij825d8s-ocaml-4.04.2.drv' failed with exit code 2
[31;1merror:[0m build of '/nix/store/ink9f6z9z231c4iwcgkbxnfpij825d8s-ocaml-4.04.2.drv' failed
```
</p></details>

--- a/pkgs/top-level/ocaml-packages.nix
+++ b/pkgs/top-level/ocaml-packages.nix
@@ -1046,5 +1046,10 @@ in rec
 
   ocamlPackages_latest = ocamlPackages_4_06;
 
-  ocamlPackages = ocamlPackages_4_05;
+  ocamlPackages =
+    # OCaml 4.05 is broken on aarch64
+    if system == "aarch64-linux" then
+      ocamlPackages_4_06
+    else
+      ocamlPackages_4_05;
 }

